library/item/item_group/src/main/java/org/quiltmc/qsl/item/group/impl/ItemGroupTextureIcon.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.group.impl;
+
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+import org.quiltmc.qsl.item.group.api.ItemGroupIcon;
+import org.quiltmc.qsl.resource.loader.api.client.ClientResourceLoaderEvents;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.DrawableHelper;
+import net.minecraft.client.texture.MissingSprite;
+import net.minecraft.client.texture.TextureManager;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+
+@ApiStatus.Internal
+public final class ItemGroupTextureIcon implements ItemGroupIcon, ClientResourceLoaderEvents.EndResourcePackReload {
+	// NB: These have to be suppliers since these objects are created before minecraft has a instance?
+	private final Supplier<TextureManager> textureManagerSupplier = () -> MinecraftClient.getInstance().getTextureManager();
```

Comment:
Is there any reason to have a field for these at all if you're just getting them every time you use them anyways?